### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by anthony-pastor** [DataDog/datadog-agent#5201]
We observed datadog-agent pod crashing randomly with this error message:

```
panic: runtime error: invalid memory address or nil pointer dereference
--
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x2ae25a8]goroutine 430056 [running]:
github.com/DataDog/datadog-agent/pkg/logs/decoder.NewDecoderWithEndLineMatcher(0x0, 0x3aac500, 0xc001d5b330, 0x3aac540, 0xc001d5b340, 0xc00420dc80)
/.omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/logs/decoder/decoder.go:70 +0x68
github.com/DataDog/datadog-agent/pkg/logs/input/docker.InitializeDecoder(...)
/.omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/logs/input/docker/matcher.go:17
github.com/DataDog/datadog-agent/pkg/logs/input/docker.NewTailer(0xc00065dee0, 0xc002433a49, 0x40, 0x0, 0xc000156d20, 0xc000640ba0, 0x0)
/.omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/logs/input/docker/tailer.go:64 +0xbc
github.com/DataDog/datadog-agent/pkg/logs/input/docker.(*Launcher).restartTailer(0xc0006b4990, 0xc002433a49, 0x40)
/.omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/logs/input/docker/launcher.go:255 +0x156
created by github.com/DataDog/datadog-agent/pkg/logs/input/docker.(*Launcher).run
/.omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/logs/input/docker/launcher.go:153 +0x3ab
```

Our env is:

- Datadog Chart: 2.0.14

- Platform: Google Kubernetes Engine cluster, version: 1.15.8-gke.3

- Chart values:

```
datadog:
  dogstatsd:
    useSocketVolume: true
  collectEvents: true
  leaderElection: true
  logs:
    enabled: true
    containerCollectAll: true
  processAgent:
    enabled: true
  tags:
    - environment:preview
clusterAgent:
  enabled: true
  podAnnotations:
    sidecar.istio.io/inject: "false"
agents:
  podAnnotations:
    sidecar.istio.io/inject: "false"
  tolerations:
    - effect: NoExecute
      operator: Exists
    - effect: NoSchedule
      operator: Exists
  updateStrategy:
    type: RollingUpdate
    rollingUpdate:
      maxUnavailable: "20%"
  containers:
    agent:
      resources:
        requests:
          cpu: 200m
          memory: 256Mi
        limits:
          cpu: "1"
          memory: 512Mi
```

**Comment by MichielDeMey** [comment]
We hit the exact same issue on our cluster, looks like it has something to do with the log parsing from the Docker containers.

To mitigate this, we've disabled log collection from Docker containers in the Helm chart for now.
`containerCollectAll: false`

**Comment by olivielpeau** [comment]
Thanks for reporting this issue @anthony-pastor and @MichielDeMey, we'll look into it.

Could you confirm with which Agent version(s) you're experiencing this?

**Comment by anthony-pastor** [comment]
@olivielpeau `root@datadog-jkfqk:/opt/datadog-agent/bin# ./agent/agent version
Agent 7.18.1 - Commit: 84d743e - Serialization version: 4.25.0 - Go version: go1.12.9
`

**Comment by prognant** [comment]
Hello @anthony-pastor, 
The PR #4885 has been merged into master. While it does not specifically target the issue your reported, we were able to exactly reproduce the issue and after thorough analysis of the aforementioned PR and the issue, it appears that the following safety check has been added :
https://github.com/DataDog/datadog-agent/blob/d8081d5e508872722634941dd71ffd389c58da00/pkg/logs/input/docker/launcher.go#L246-L249
It prevents the crash from happening. The PR is included in the 7.19 release.
